amazon-kinesis-client/src/main/java/software/amazon/kinesis/retrieval/DataFetcherProviderConfig.java

```diff
@@ -1,26 +1,48 @@
+/*
+ * Copyright 2019 Amazon.com, Inc. or its affiliates.
+ * Licensed under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
 package software.amazon.kinesis.retrieval;
 
-import lombok.Data;
-import lombok.NonNull;
+import java.time.Duration;
 import software.amazon.kinesis.common.StreamIdentifier;
 import software.amazon.kinesis.metrics.MetricsFactory;
 
+public interface DataFetcherProviderConfig {
```

Comment:
@ashwing 
Should we expose the DataFetcherProviderConfig and have that be used instead of the current KinesisDataFetcherProviderConfig if DataFetcherProviderConfig is provided by the user?

Reply:
we will expose only the interface DataFetcherProviderConfig to the end user. The actual concrete class we give to the customer doesn't matter to them.